HopsanGUI/PlotTab.cpp

```diff
@@ -1366,12 +1380,51 @@ void PlotTab::openFrequencyAnalysisDialog(PlotCurve *pCurve)
     QDialog *pDialog = new QDialog(this);
     pDialog->setWindowTitle("Generate Frequency Spectrum");
 
-    QCheckBox *pLogScaleCheckBox = new QCheckBox("Use log scale");
+    QCheckBox *pLogScaleCheckBox = new QCheckBox("Use log scale", this);
     pLogScaleCheckBox->setChecked(true);
 
-    QCheckBox *pPowerSpectrumCheckBox = new QCheckBox("Power spectrum");
+    QCheckBox *pPowerSpectrumCheckBox = new QCheckBox("Power spectrum", this);
     pPowerSpectrumCheckBox->setChecked(false);
 
+    QGroupBox *pWindowingGroupBox = new QGroupBox("Windowing", pDialog);
+    QGridLayout *pWindowingLayout = new QGridLayout(pWindowingGroupBox);
+
+    QLabel *pWindowingLabel = new QLabel("Windowing function:",pDialog);
+    QComboBox *pWindowingComboBox = new QComboBox(pDialog);
+    pWindowingComboBox->addItems(QStringList() << "Rectangular" << "Hann Function");
+    pWindowingComboBox->setCurrentIndex(1);
+
+    double minX = pCurve->getSharedTimeOrFrequencyVariable()->minOfData();
+    double maxX = pCurve->getSharedTimeOrFrequencyVariable()->maxOfData();
+
+    QLabel *pMinTimeLabel = new QLabel("Min time: ", pDialog);
+
+    mpWindowingMinTimeSpinBox = new QDoubleSpinBox(this);
+    mpWindowingMinTimeSpinBox->setMinimum(minX);
+    mpWindowingMinTimeSpinBox->setMaximum(maxX);
+    mpWindowingMinTimeSpinBox->setValue(minX);
+    mpWindowingMinTimeSpinBox->setDecimals(2-qFloor(log10(maxX-minX)));
+    mpWindowingMinTimeSpinBox->setSingleStep(pow(10,qFloor(log10(maxX-minX))-1));
+
+    QLabel *pMaxTimeLabel = new QLabel("Max time: ", pDialog);
+
+    mpWindowingMaxTimeSpinBox = new QDoubleSpinBox(this);
+    mpWindowingMaxTimeSpinBox->setMinimum(minX);
+    mpWindowingMaxTimeSpinBox->setMaximum(maxX);
+    mpWindowingMaxTimeSpinBox->setValue(maxX);
+    mpWindowingMaxTimeSpinBox->setDecimals(2-qFloor(log10(maxX-minX)));
```

Comment:
What would happen if maxX-minX == 0 ?
I am thinking about the log10() expression
Nut sure if that can happen

Reply:
It can happen if you use only one log sample in the simulation (unlikely), or if you make an X-Y plot with a constant variable as X (little less unlikely). I fixed it just be be on the safe side.